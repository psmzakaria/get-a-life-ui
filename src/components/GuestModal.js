import React, { Component } from "react";
import { Modal, Button } from "semantic-ui-react";
import { API_URL } from "../utils/configVar";
import InfiniteCalendar, {
  Calendar,
  defaultMultipleDateInterpolation,
  withMultipleDates
} from "react-infinite-calendar";
import "react-infinite-calendar/styles.css";
import format from "date-fns/format";
import EventCard from "./EventCard";

class GuestModal extends Component {
  constructor() {
    super();
    this.state = {
      dates: [],
      dummy: [],
      modalOpen: false
    };
  }

  handleOpen = () => {
    this.setState({
      modalOpen: true
    });
  };

  handleClose = () => {
    this.setState({
      modalOpen: false
    });
  };

  handleAcceptSubmit = async event => {
    event.preventDefault();
    const eventId = this.props.event._id;
    const response = await fetch(`${API_URL}/events/${eventId}/rsvp`, {
      method: "PUT",
      credentials: "include",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        status: "accepted",
        availableDates: this.state.dates
      })
    });
    if (response.ok) {
      this.handleClose();
      this.props.getUserData();
      console.log("updated successfully");
    }
  };

  handleCalendarSelect = date => {
    const formattedDate = format(date, "YYYYMMDD");
    const updatedDates = defaultMultipleDateInterpolation(
      formattedDate,
      this.state.dates
    );
    this.setState({
      dates: updatedDates
    });
  };

  render() {
    const { title, hostId, proposedDates, description } = this.props.event;
    const { username: hostName } = hostId;
    const firstDate = proposedDates[0];
    const lastDate = proposedDates[proposedDates.length - 1];
    const minDate = new Date(
      Date.UTC(
        firstDate.substr(0, 4),
        firstDate.substr(4, 2) - 1,
        firstDate.substr(6, 2)
      )
    );
    const maxDate = new Date(
      Date.UTC(
        lastDate.substr(0, 4),
        lastDate.substr(4, 2) - 1,
        lastDate.substr(6, 2)
      )
    );

    return (
      <div className="esnap-container-size">
        <Modal
          trigger={
            <EventCard
              handleOpen={this.handleOpen}
              event={this.props.event}
              status={this.props.status}
            />
          }
          open={this.state.modalOpen}
          onClose={this.handleClose}
          closeIcon
        >
          <Modal.Header>{title}</Modal.Header>
          <Modal.Content>
            <Modal.Description>
              {hostName}
              <br />
              <strong> Description:</strong> {description}
              <br />
              <div>
                <InfiniteCalendar
                  autoFocus={false}
                  width={
                    window.innerWidth <= 650
                      ? window.innerWidth * 0.85
                      : window.innerWidth * 0.6
                  }
                  height={window.innerHeight - 450}
                  displayOptions={{
                    showHeader: false
                  }}
                  min={minDate}
                  minDate={minDate}
                  max={maxDate}
                  maxDate={maxDate}
                  Component={withMultipleDates(Calendar)}
                  selected={this.state.dummy}
                  interpolateSelection={defaultMultipleDateInterpolation}
                  onSelect={this.handleCalendarSelect}
                  theme={{
                    selectionColor: "rgb(146, 118, 255)",
                    textColor: {
                      default: "#333",
                      active: "#FFF"
                    },
                    weekdayColor: "rgb(146, 118, 255)",
                    headerColor: "rgb(127, 95, 251)",
                    floatingNav: {
                      background: "rgba(81, 67, 138, 0.96)",
                      color: "#FFF",
                      chevron: "#FFA726"
                    }
                  }}
                />
              </div>
              <br />
            </Modal.Description>
          </Modal.Content>
          <Modal.Actions>
            <Button
              type="submit"
              content="Accept"
              onClick={event => this.handleAcceptSubmit(event)}
            />
          </Modal.Actions>
        </Modal>
      </div>
    );
  }
}

export default GuestModal;
