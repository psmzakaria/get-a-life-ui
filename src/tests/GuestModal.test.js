import React from "react";
import GuestModal from "../components/GuestModal";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders the create component of the app ", () => {
  const renderer = new ShallowRenderer();
  const event = {
    title: "birthday",
    hostId: {
      username: "lala"
    },
    proposedDates: ["20180813", "20180814", "20180815"]
  };

  renderer.render(<GuestModal key={1} event={event} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
