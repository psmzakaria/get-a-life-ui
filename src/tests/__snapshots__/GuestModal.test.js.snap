// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the create component of the app  1`] = `
<div
  className="esnap-container-size"
>
  <Modal
    centered={true}
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    trigger={
      <Card
        color="violet"
        onClick={[Function]}
      >
        <CardContent>
          <CardHeader>
            birthday
          </CardHeader>
          <CardMeta>
            <span>
              Host: 
              lala
            </span>
          </CardMeta>
          <CardDescription>
             
            <Icon
              as="i"
              name="list"
              size="small"
            />
             KIV - 
          </CardDescription>
        </CardContent>
        <CardContent
          extra={true}
        />
      </Card>
    }
  >
    <ModalHeader>
      birthday
    </ModalHeader>
    <ModalContent>
      <ModalDescription>
        lala
        <br />
        <strong>
           Description:
        </strong>
         
        <br />
        <div>
          <DefaultCalendar
            Component={[Function]}
            autoFocus={false}
            displayOptions={
              Object {
                "showHeader": false,
              }
            }
            height={318}
            interpolateSelection={[Function]}
            max={2018-08-15T00:00:00.000Z}
            maxDate={2018-08-15T00:00:00.000Z}
            min={2018-08-13T00:00:00.000Z}
            minDate={2018-08-13T00:00:00.000Z}
            onSelect={[Function]}
            selected={Array []}
            theme={
              Object {
                "floatingNav": Object {
                  "background": "rgba(81, 67, 138, 0.96)",
                  "chevron": "#FFA726",
                  "color": "#FFF",
                },
                "headerColor": "rgb(127, 95, 251)",
                "selectionColor": "rgb(146, 118, 255)",
                "textColor": Object {
                  "active": "#FFF",
                  "default": "#333",
                },
                "weekdayColor": "rgb(146, 118, 255)",
              }
            }
            width={614.4}
          />
        </div>
        <br />
        <Button
          as="button"
          floated="right"
          onClick={[Function]}
          role="button"
          type="submit"
        >
          Accept
        </Button>
      </ModalDescription>
    </ModalContent>
  </Modal>
</div>
`;
